### DISCUSSION TITLE
https://api.nn.ci/alist/ali_open/token连接被拒绝

### DISCUSSION THREAD

**Discussion by shinegogo** [AlistGo/alist#3759] [Q&A]
版本如下：
$ alist version
Built At: 2023-03-06 13:06:25 +0000
Go Version: go1.19.6 linux/amd64
Author: Xhofe <i@nn.ci>
Commit ID: dd4c973
Version: v3.13.1
WebVersion: 3.13.0

log如下：
$ alist server
INFO[2023-03-07 10:28:00] reading config file: data/config.json
INFO[2023-03-07 10:28:00] load config from env with prefix: ALIST_
INFO[2023-03-07 10:28:00] init logrus...
INFO[2023-03-07 10:28:00] start server @ 0.0.0.0:5244
2023/03/07 10:28:00.394498 ERROR RESTY Post "https://api.nn.ci/alist/ali_open/token": dial tcp: lookup api.nn.ci on [::1]:53: read udp [::1]:56855->[::1]:53: read: connection refused, Attempt 1
INFO[2023-03-07 10:28:00] qbittorrent not ready.
2023/03/07 10:28:00.500661 ERROR RESTY Post "https://api.nn.ci/alist/ali_open/token": dial tcp: lookup api.nn.ci on [::1]:53: read udp [::1]:53823->[::1]:53: read: connection refused, Attempt 2
2023/03/07 10:28:00.700996 ERROR RESTY Post "https://api.nn.ci/alist/ali_open/token": dial tcp: lookup api.nn.ci on [::1]:53: read udp [::1]:49638->[::1]:53: read: connection refused, Attempt 3
2023/03/07 10:28:00.962268 ERROR RESTY Post "https://api.nn.ci/alist/ali_open/token": dial tcp: lookup api.nn.ci on [::1]:53: read udp [::1]:45698->[::1]:53: read: connection refused, Attempt 4
ERRO[2023-03-07 10:28:00] failed get enabled storages: Post "https://api.nn.ci/alist/ali_open/token": dial tcp: lookup api.nn.ci on [::1]:53: read udp [::1]:45698->[::1]:53: read: connection refused
failed init storage
github.com/alist-org/alist/v3/internal/op.initStorage
        /home/runner/work/alist/alist/internal/op/storage.go:99
github.com/alist-org/alist/v3/internal/op.LoadStorage
        /home/runner/work/alist/alist/internal/op/storage.go:80
github.com/alist-org/alist/v3/internal/bootstrap.LoadStorages.func1
        /home/runner/work/alist/alist/internal/bootstrap/storage.go:20
runtime.goexit
        /opt/hostedtoolcache/go/1.19.6/x64/src/runtime/asm_arm64.s:1172
INFO[2023-03-07 10:28:01] Aria2 not ready.

**Comment by tmzg0000** [comment]
我也有此问题，后来靠开启ipv6解决，就是docker里的网络用host模式同时主机开启ipv6  希望能帮到你

**Comment by SpeedPartner** [comment]
代理最好不要用国外的远程DNS,或者把阿里云盘的相关域名放到直连名单.
用国外的DNS连阿里云盘的域名就会这样,毕竟阿里云盘还没在国外运营.

**Comment by songxiaocheng** [comment]
将配置里的"Oauth token url"项中的 https://api.nn.ci 改成 https://api-cf.nn.ci 或者 https://api.xhofe.top 试试。我用这种方式解决了该问题。

注意这里api.nn.ci不是阿里云的域名而是AList的域名，这个是日本节点。另外两个分别是Cloudflare的和国内的。文档中我暂时没找到关于这一点的说明，不过如果进入文档的Storage部分，左侧会有个下拉框，可以看到这三个不同的域名。

**Reply by xuezhonghai** [reply]
感谢，解决了不能访问这个链接的问题

**Reply by BruceoZhang** [reply]
感谢！第二个有用，第一个失败

**Reply by zhushen12580** [reply]
哪个配置

**Reply by songxiaocheng** [reply]
> 哪个配置

英文/中文界面：`AList Manage`/`AList管理` -> `Storages`/`存储`。如果本地部署且没改端口的话网址是 http://127.0.0.1:5244/@manage/storages
对有问题的AliyunDrive，点`Edit`/`编辑`，找到`Oauth token url`/`Oauth令牌链接`。
